test/e2e/categories_test.go

```diff
@@ -0,0 +1,186 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+
+	infrav1 "github.com/nutanix-cloud-native/cluster-api-provider-nutanix/api/v1beta1"
+)
+
+const (
+	defaultClusterCategoryKeyPrefix           = "kubernetes-io-cluster"
+	defaultClusterCategoryValue               = "owned"
+	defaultNonExistingAdditionalCategoryKey   = "nonExistingCategoryKeyCAPX"
+	defaultNonExistingAdditionalCategoryValue = "nonExistingCategoryValueCAPX"
+)
+
+var _ = Describe("Nutanix categories [PR-Blocking]", func() {
+	var (
+		namespace        *corev1.Namespace
+		specName         = "cluster-categories"
+		clusterName      string
+		clusterResources *clusterctl.ApplyClusterTemplateAndWaitResult
+		cancelWatches    context.CancelFunc
+	)
+	BeforeEach(func() {
+		clusterName = generateTestClusterName(specName)
+		clusterResources = new(clusterctl.ApplyClusterTemplateAndWaitResult)
+		Expect(bootstrapClusterProxy).NotTo(BeNil(), "BootstrapClusterProxy can't be nil")
+		namespace, cancelWatches = setupSpecNamespace(ctx, specName, bootstrapClusterProxy, artifactFolder)
+	})
+	AfterEach(func() {
```

Comment:
Nit: be consistent with whitespace after Ginkgo blocks.

Reply:
Modified